presto-orc/src/test/java/com/facebook/presto/orc/OrderTrackingOrcFilter.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.orc;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public interface OrderTrackingOrcFilter
+        extends TupleDomainFilter
+{
+    /**
+     * This is a tracking interface to show that this filter implements orderTracking
+     */
+
+    /**
+     * @return the associated orderChecker object
+     */
+    OrderChecker getOrderChecker();
+
+    /**
+     * Set a variable corresponding to which column this filter covers.
+     */
+    void register(int column);
+
+    class OrderChecker
+    {
+
+        private List<Integer> columnOrdering;
+        private List<Integer> desiredOrder;
+        private boolean started;
+        private boolean ended;
+
+        OrderChecker()
```

Comment:
I think this class can be simplified to just a `call(column)` and `assertOrder()` methods.

```
   class TupleDomainFilterOrderChecker
    {
        private final List<Integer> expectedOrder;
        private final List<Integer> columnOrdering = new ArrayList<>();

        TupleDomainFilterOrderChecker(List<Integer> expectedOrder)
        {
            this.expectedOrder = requireNonNull(expectedOrder, "expectedOrder is null");
        }

        public void call(int column)
        {
            if (!columnOrdering.contains(column)) {
                columnOrdering.add(column);
            }
        }

        public void assertOrder()
        {
            assertEquals(columnOrdering, expectedOrder);
        }
    }
```